packages-ts/starknet-gauntlet-ocr2/src/commands/ocr2/setConfig.ts

```diff
@@ -123,8 +126,32 @@ const makeContractInput = async (
   return [oracles, input.f, onchainConfig, 2, bytesToFelts(offchainConfig)]
 }
 
-// TODO: beforeExecute attempt to deserialize offchainConfig to check for validity
-// TODO: diff onchain config vs proposed config
+const beforeExecute: BeforeExecute<SetConfigInput, ContractInput> = (
+  context,
+  input,
+  deps,
+) => async () => {
+  deps.logger.loading(`Executing ${context.id} from contract ${context.contractAddress}`)
+
+  const { offchainConfig } = await encoding.serializeOffchainConfig(
+    input.user.offchainConfig,
+    input.user.secret,
+  )
+  const newOffchainConfig = encoding.deserializeConfig(offchainConfig)
+
+  const eventData = await getLatestOCRConfigEvent(context.provider, context.contractAddress)
+  if (eventData.length === 0) {
+    deps.logger.info('No previous config found, review the offchain config below:')
+    console.log(newOffchainConfig)
```

Comment:
Just curious, why do we need to use different logging methods here? Sorry if it's too trivial as I'm not yet familiar with this codebase.

Reply:
good catch, this slipped through 😅